utils/githooks/README.md

```diff
@@ -1,19 +1,18 @@
 # Git hooks
 
-This directory contains githooks that can be employed locally to
-automate things like copyright updates on modified files
+This directory contains a githooks framework that can be
+employed locally to automate things like copyright updates
+on modified files.
+
+It supports adding local hooks by placing your custom hook in
+utils/githooks/<hook>.d/<num>-<name> files.
 
 To use the pre-commit hook here, do the following locally
 
 ```
-
-cat << EOF 2>&1 > .git/hooks/pre-commit
-#!/bin/sh
-
-# Invoke the githook for copyright validation
-exec utils/githooks/pre-commit 1>&2
-EOF
-
-chmod 700 .git/hooks/pre-commit
-
+for d in utils/githooks/*.d; do
+    h=${d%.d}
+    h=${h##*/}
+    ln -s ../../utils/githooks/$h .git/hooks/$h
+done
```

Comment:
If this becomes more common (or required), maybe we should have an executable script do this for us. We could even have a hook to relink all the new hooks!